docs/dev-sdk.md

```diff
@@ -1,59 +1,199 @@
 # SDK Development
 This document will highlight how to add new APIs to the OpenStorage SDK.
 
-## SDK Requirements
+## OpenStorage SDK Guide
+
+### Protocol Buffer Style Guide
+
+The SDK follows the [Protocol Buffer Style Guide](https://developers.google.com/protocol-buffers/docs/style). From the guide:
+
+>Note that protocol buffer style has evolved over time, so it is likely that you will see .proto files written in different conventions or styles. Please respect the existing style when you modify these files. Consistency is key. However, it is best to adopt the current best style when you are creating a new .proto file.
+
+Therefore, for any new messages, enums, etc, please follow the style guide.
+
+### OpenStorage Style
+
 All SDK APIs and values must satisfy by the following:
 
-* Must be readable
-    * SDK APIs and values must be concrete clear values
-    * `string` values are for ids or strings. They are not meant to marshal information as a `yaml`, create a concrete _message_ instead.
+#### Version
+
+Any changes to the protocol must bump the version by one. On the _master_ branch, the _minor_ number is bumped. On _release_ branches, the _patch_ number is bumped.
+
+#### Readability 
+
+**APIs Must be readable**. SDK APIs and values must be concrete clear values. They are used not just by Portworx, but also non-Portworx developers which do not have an understanding of the internals of the Portworx cluster.
+
+#### `string` values.
```

Comment:
maybe include a note on avoid opaque map[string]string options when possible? I remember you mentioning this in PR reviews.